### PR TITLE
[EBPF-575] gpu: add more fatbin samples for tests

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Adds an auto-generated CUDA executable to stress test the fatbin parser.

### Additional Notes

I profiled the code using this new file, there doesn't seem to be any obvious/easy improvements to be made. Most of the time is spent reading the file from disk, and most of the memory usage comes from the need to read the fatbin payload into memory. Being able to read both compressed and uncompressed payloads without copying everything would be useful to reduce the memory usage.

### CODE REVIEW

File: pkg/gpu/cuda/fatbin_test.go [hunk 1]
```diff
@@ -19,18 +21,18 @@ import (
 )
 
 // The test data is a CUDA fatbin file compiled with the Makefile present in the same directory.
```
Comment (val06) [inline pkg/gpu/cuda/fatbin_test.go:23]: _nit:_ maybe update the comment

**Review by val06 — APPROVED** [review]
small inline comment + can you add somewhere in code (for your preference) a description of what is the "heavy-sample"? (size wise, or any other parameters that are influencing the code under test)

**Comment by gjulianm** [comment]
Benchmark results:

``` 
BenchmarkParseFatbinFromPath/heavy-sample-8                   26          43142327 ns/op         7744523 B/op     118793 allocs/op
```

File: pkg/gpu/cuda/fatbin_test.go (hunk 1 above)
Comment (gjulianm) [reply pkg/gpu/cuda/fatbin_test.go:23]: Updated

**Pull request merged**